### PR TITLE
Use safer dedupe for config

### PR DESCRIPTION
**Pull request by howardjohn**
This is just appending them, there is no guarantee of conflicts being avoided

**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/config/aggregate/config.go [hunk 1]
```diff
@@ -110,13 +111,12 @@ func (cr *store) List(typ config.GroupVersionKind, namespace string) []config.Co
 	}
 	var configs []config.Config
 	// Used to remove duplicated config
-	configMap := sets.New[string]()
+	configMap := sets.New[types.NamespacedName]()
 
 	for _, store := range cr.stores[typ] {
 		storeConfigs := store.List(typ, namespace)
 		for _, cfg := range storeConfigs {
-			key := cfg.GroupVersionKind.Kind + cfg.Namespace + cfg.Name
```
Comment (howardjohn) [inline pilot/pkg/config/aggregate/config.go:118]: Not: Kind is not needed since List already is for a single GVK

**Review by hzxuzhonghu — APPROVED** [review]

File: pilot/pkg/config/aggregate/config.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/config/aggregate/config.go:118]: is namespace needed?

Comment (hzxuzhonghu) [reply pilot/pkg/config/aggregate/config.go:118]: Sure, List() can list all namespace configs

**Pull request merged**